src/esmockCache.js

```diff
@@ -7,6 +7,8 @@ const esmockCache = {
 }
 
 const esmockTreeIdSet = (key, keylong) => (
+  (typeof global.postMessageEsmk === 'function')
```

Comment:
'old' and 'new' variants are mutual exclusion - the old one doesn't work with dedicated thread, and the new one doesn't exist if globalPreload hook is not supported.

I propose to split those branches explicitly and choose the way based on postMessageEsmk existance.

Reply:
its _almost_ working we just need to disable test-tsm, which fails with this message
```bash
error: Invalid version: "21.0.0-nightly202305158b3777d0c8"
```

Reply:
It seems a split is not needed, but perhaps we should try using an unary here or some other mechanism for reducing the number of definitions created.  I would prefer to make such changes in a follow up PR just to this PR along and publish it so esmock will work with the latest node versions again

Reply:
I'll try pushing up changes with an unary to resolve this discussion

Reply:
this change caused the tests to fail https://github.com/iambumblehead/esmock/actions/runs/4990343286/jobs/8935419674?pr=199 I have an idea around why it happens but am sleepy its 2:45am here (reverting)
```diff
diff --git a/src/esmockCache.js b/src/esmockCache.js
index 51c625b..0420787 100644
--- a/src/esmockCache.js
+++ b/src/esmockCache.js
@@ -8,8 +8,8 @@ const esmockCache = {
 
 const esmockTreeIdSet = (key, keylong) => (
   typeof global.postMessageEsmk === 'function'
-    && global.postMessageEsmk({ key, keylong }),
-  global.mockKeys[String(key)] = keylong)
+    ? global.postMessageEsmk({ key, keylong })
+    : global.mockKeys[String(key)] = keylong)
 
 const esmockTreeIdGet = key => (
   global.mockKeys[String(key)])

```

Reply:
Politely, I'd like to mark this as resolved. The change was requested when tests were still failing but now they pass (when the diff from the previous comment is not applied).